src/ifaces.ts

```diff
@@ -55,12 +55,26 @@ export interface TransactionOptions {
   readonly?: boolean;
 }
 
-export interface Connection {
+export interface ReadOnlyExecutor {
   execute(query: string): Promise<void>;
   query(query: string, args?: QueryArgs): Promise<Set>;
   queryJSON(query: string, args?: QueryArgs): Promise<string>;
   queryOne(query: string, args?: QueryArgs): Promise<any>;
   queryOneJSON(query: string, args?: QueryArgs): Promise<string>;
+}
+
+export const allow_modifications = Symbol();
+
+interface Modifiable {
```

Comment:
MutatingExecutor?

Reply:
We use "modifications" rather than "mutations" in RFC1004. Should we change that?

Also it's not executor, it becomes executor when combining `ReadOnlyExecutor & Modifiable`. So this is kinda "aspect" kind of type. But it's internal anyway, so I'm not going to fight for the name too much.